pkg/hintrunner/zero/zerohint_dictionaries.go

```diff
@@ -33,3 +35,61 @@ func newSquashDictInnerAssertLenKeysHint() hinter.Hinter {
 func createSquashDictInnerAssertLenKeysHinter() (hinter.Hinter, error) {
 	return newSquashDictInnerAssertLenKeysHint(), nil
 }
+
+// SquashDictInnerNextKey hint retrieves the next key for processing during
+// dictionary squashing after checking that the array of keys is not empty
+//
+// `newSquashDictInnerNextKeyHint` takes 1 operander as argument
+//   - `next_key` variable will be assigned to the next key in `keys`
+func newSquashDictInnerNextKeyHint(nextKey hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SquashDictInnerNextKey",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> assert len(keys) > 0, 'No keys left but remaining_accesses > 0.'
+			//> ids.next_key = key = keys.pop()
+
+			keys_, err := ctx.ScopeManager.GetVariableValue("keys")
+			if err != nil {
+				return err
+			}
+
+			keys := keys_.([]f.Element)
+			if len(keys) == 0 {
+				return fmt.Errorf("no keys left but remaining_accesses > 0")
+			}
+
+			newKey, err := utils.Pop(&keys)
+			if err != nil {
+				return err
+			}
+
+			err = ctx.ScopeManager.AssignVariable("keys", keys)
+			if err != nil {
+				return err
+			}
```

Comment:
In case you're unaware, this type of patter can be merged in Go with the following syntax:

```
if err := ctx.ScopeManager.Smth(...); err != nil {
    return err
}
```

Just another way of writing it, you pick whatever you like the most

Reply:
thanks!